test/unit/test-amp-story-player.js

```diff
@@ -160,19 +177,32 @@ describes.realWin('AmpStoryPlayer', {amp: false}, env => {
     buildStoryPlayer();
 
     messagingMock.expects('registerHandler').withArgs('selectDocument');
+    messagingMock.expects('registerHandler').withArgs('touchstart');
+    messagingMock.expects('registerHandler').withArgs('touchmove');
+    messagingMock.expects('registerHandler').withArgs('touchend');
     messagingMock.expects('setDefaultHandler');
 
-    await player.buildCallback();
+    await manager.loadPlayers();
   });
 
   it('should navigate to next story when the last page of a story is tapped', async () => {
     buildStoryPlayer(2);
 
-    await player.buildCallback();
-    player.layoutCallback();
+    await manager.loadPlayers();
 
     const fakeData = {next: true};
-    fireHandler('selectDocument', fakeData);
+    fireHandler['selectDocument']('selectDocument', fakeData);
+
+    const iframes = playerEl.shadowRoot.querySelectorAll('iframe');
+    expect(iframes[0].getAttribute('i-amphtml-iframe-position')).to.eql('-1');
+    expect(iframes[1].getAttribute('i-amphtml-iframe-position')).to.eql('0');
+  });
+
+  it('should navigate when swiping', async () => {
```

Comment:
Can you add a test when swiping on a first (or last) story?

Reply:
Done.